### PR TITLE
add `true()` construction; remove now-redundant `true([F])` construction

### PR DESCRIPTION
**Pull request by apoelstra**

### CODE REVIEW

**Comment by apoelstra** [comment]
cc @sanket1729 for code review, @sipa for concept ACK

File: src/policy/mod.rs [hunk 1]
```diff
@@ -348,6 +352,7 @@ impl<P> AbstractPolicy<P> {
     /// returns `None`.
     pub fn before_time(mut self, time: u32) -> Option<AbstractPolicy<P>> {
         self = match self {
+            AbstractPolicy::True => AbstractPolicy::True,
```
Comment (sanket1729) [inline src/policy/mod.rs]: I think there is no need to add a special case here, this is already covered in the last case, 
x => x line 385

File: src/miniscript/astelem.rs [hunk 2]
```diff
@@ -112,6 +113,7 @@ impl<P> AstElem<P> {
     /// Does the element satisfy the "expression" calling convention?
     pub fn is_e(&self) -> bool {
         match *self {
+            AstElem::True => true,
```
[the remark this answers is not in the document]
Comment (apoelstra) [reply src/miniscript/astelem.rs]: Yep, good catch.

File: src/miniscript/astelem.rs [hunk 3]
```diff
@@ -1043,6 +1041,7 @@ impl<P: ToPublicKey> AstElem<P> {
     pub fn max_dissatisfaction_size(&self, one_cost: usize) -> usize {
         assert!(self.is_e() || self.is_w());
         match *self {
+            AstElem::True => 0,
```
[the remark this answers is not in the document]
Comment (apoelstra) [reply src/miniscript/astelem.rs]: It can't be `e` or `w` you mean - it's definitely `f`

[the remark this answers is not in the document]
Comment (sanket1729) [reply src/miniscript/astelem.rs]: Yeah, sorry.  can't be `e` or `w`.

**Comment by apoelstra** [comment]
Deleted all the lines with nits.

**Comment by sanket1729** [comment]
tested fb19d216382db6a77dc867770e27d9c207733933

**Comment by apoelstra** [comment]
tested ACK? :)

**Comment by sanket1729** [comment]
Yeah, Ack :)

**Pull request merged**